code/modules/overmap/exoplanets/turfs.dm

```diff
@@ -0,0 +1,229 @@
+/turf/simulated/floor/exoplanet
+	name = "space land"
+	icon = 'icons/turf/desert.dmi'
+	icon_state = "desert"
+	has_resources = 1
+	footstep_sound = /decl/sound_category/asteroid_footstep
+	var/diggable = 1
+	var/dirt_color = "#7c5e42"
+
+/turf/simulated/floor/exoplanet/New()
+	if(current_map.use_overmap)
+		var/obj/effect/overmap/visitable/sector/exoplanet/E = map_sectors["[z]"]
+		if(istype(E))
+			if(E.atmosphere)
+				temperature = E.atmosphere.temperature
+			else
+				temperature = T0C
+			//Must be done here, as light data is not fully carried over by ChangeTurf (but overlays are).
+			set_light(E.lightlevel, 0.1, 2)
+			if(E.planetary_area && istype(loc, world.area))
+				ChangeArea(src, E.planetary_area)
+	..()
+
+/turf/simulated/floor/exoplanet/attackby(obj/item/C, mob/user)
+	if(diggable && istype(C,/obj/item/shovel))
+		visible_message("<span class='notice'>\The [user] starts digging \the [src]</span>")
+		if(do_after(user, 50))
+			to_chat(user,"<span class='notice'>You dig a deep pit.</span>")
+			new /obj/structure/pit(src)
+			diggable = 0
+		else
+			to_chat(user,"<span class='notice'>You stop shoveling.</span>")
+	else if(istype(C, /obj/item/stack/tile))
+		var/obj/item/stack/tile/T = C
+		if(T.use(1))
+			playsound(src, 'sound/items/Deconstruct.ogg', 80, 1)
+			ChangeTurf(/turf/simulated/floor, FALSE, FALSE, TRUE)
```

Comment:
shouldn't this adopt the turf type of the stack

Reply:
Don't see a variable that lets me do this